_published/0014-test-results-handling.md

```diff
@@ -0,0 +1,274 @@
+
+# Update the following (it's YAML syntax)
+pr_id: 14 # Update this with PR number/ID. No leading zeros
+rfc_feature_name: rework-handling-of-test-results # Use kebab-case
+title: "RFC-0014: rework-handling-of-test-results" # Update this with PR number/ID and feature name. Use leading zeros
+rfc_author_username: Alexamakans
+rfc_author_name: Alexander Fougner # Or same as username, if you wish
+
+# Leave these. Collaborator changes this before merging
+impl_issue_id: 0
+impl_issue_repo: iver-wharf/wharf-api
+last_modified_date: YYYY-MM-DD
+---
+
+# {{page.title}}
+
+- RFC PR: [iver-wharf/rfcs#{{page.pr_id}}](https://github.com/iver-wharf/rfcs/pulls/{{page.pr_id}})
+- Feature name: `{{page.rfc_feature_name}}`
+- Author: {{page.rfc_author_name}} ([@{{page.rfc_author_username}}](https://github.com/{{page.rfc_author_username}}))
+- Implementation issue: [{{page.impl_issue_repo}}#{{page.impl_issue_id}}](https://github.com/{{page.impl_issue_repo}}/issues/{{page.impl_issue_id}})
+- Implementation status: ![GitHub issue state](https://img.shields.io/github/issues/detail/state/{{page.impl_issue_repo}}/{{page.impl_issue_id}}?{{page.shields_io_query_params}})
+
+## Summary
+
+The intent is to avoid filtering and parsing test result files (TRX)
+from the generated artifacts each time we want to display them or a
+summary.
+
+We also want to display more detailed test result summaries in wharf-web.
+For this we will need to also store the messages that the tests produce.
+At the very least for failed tests.
+
+## Motivation
+
+The Status field in wharf-web's project view (Failed/Completed)
+currently calls an endpoint that asks the server to fetch all TRX files
+for the build, and take the most recent one. Then it parses the file to
+see how many failed/completed tests there were.
+It does this for every build in the view, every time it is refreshed.
+This leads to unnecessary requests and computation. This, in turn, slows
+down the response times and increases the API load.
+
+Having the tests' error messages is also pretty much a necessity when trying
+to fix them. Currently, there is no way to ask the API for them.
+
+## Related GitHub issues
+
+- wharf-api [#11](https://github.com/iver-wharf/wharf-api/issues/11) [#17](https://github.com/iver-wharf/wharf-api/issues/17)
+- wharf-web [#13](https://github.com/iver-wharf/wharf-web/issues/13)
+
+## Explanation
+
+### wharf-api
+
+The `POST /build/{buildid}/artifact` endpoint handles inserting artifacts, same
+as before. However, if it's a test results file (eg: .trx), one should call the
+`POST /build/{buildid}/test-results` endpoint instead. This tells the server
+to parse them as well, creating several `TestResultDetail`s and one `TestResultSummary`
+per TRX file.
+
+The summaries get inserted into the database table `test_result_summary`.\
+The results get inserted into the database table `test_result_detail`.
+
+![Database structure](../\_assets/wharf-db-graph.png)
```

Comment:
Because `npm run lint-fix` added this backslash in the path then the URL becomes invalid as Jekyll does not unescape URLs:

![image](https://user-images.githubusercontent.com/2477952/124601171-d0bd5980-de67-11eb-9fab-743063d40e1f.png)

(for reference, `%5C` is the escape sequence for backslash `\`)

The underscore folders are special folders to Jekyll. They're considered "collections", as defined in the `_config.yml`.

Suggest renaming this new `_assets` directory to simply `assets` and then updating this path like so:

```suggestion
![Database structure](../assets/wharf-db-graph.png)
```

Reply:
Ah, I see.